include/pipe.h

```diff
@@ -0,0 +1,93 @@
+#ifndef PIPE_NETWORK_PIPE_H_
+#define PIPE_NETWORK_PIPE_H_
+
+#include "node.h"
+#include <array>
+#include <math.h>
+#include <memory>
+
+//! Pipe class
+//! \brief Class that stores the information about pipes
+class Pipe {
+
+ public:
+  // Constructor with id and node pointers
+  //! \param[in] id node id
+  //! \param[in] nodes array of node pointers
+  Pipe(unsigned id, const std::array<std::shared_ptr<Node>, 2>& nodes)
+      : id_{id}, array_node_ptr_{nodes} {}
+
+  //! Destructor
+  ~Pipe() { array_node_ptr_.fill(nullptr); }
+
+  //! Copy constructor
+  Pipe(const Pipe&) = delete;
+
+  //! Assignment operator
+  Pipe& operator=(const Pipe&) = delete;
+
+  //! Move constructor
+  Pipe(Pipe&&) = delete;
+
+  //! Return id
+  //! \retval id_ id of the pipe
+  unsigned id() const { return id_; }
+
+  //! Return array of node pointers
+  //! \retval array_node_ptr_ array of node pointers
+  std::array<std::shared_ptr<Node>, 2> array_node_ptr() const {
+    return array_node_ptr_;
+  }
+
+  //! Return discharge in the pipe
+  //! \retval discharge_ discharge
+  unsigned discharge() const { return discharge_; }
+
+  //! Assign radius of the pipe
+  //! \param[in] radius radius of the pipe
+  void radius(double radius) { radius_ = radius; }
+
+  //! Assign length of the pipe
+  //! \param[in] length of the pipe
+  // void length(double length) { length_ = length; }
```

Comment:
You shouldn't assign the length of the pipe, rather calculate it. You have all the necessary information to calculate the length of the pipe